neqo-transport/tests/server.rs

```diff
@@ -210,15 +210,14 @@ fn same_initial_after_connected() {
     let server_initial = server.process(client_initial.as_dgram_ref(), now()).dgram();
     assert!(server_initial.is_some());
     complete_connection(&mut client, &mut server, server_initial);
-    // This removes the connection from the active set until something happens to it.
-    assert_eq!(server.active_connections().len(), 0);
+    assert_eq!(server.active_connections().len(), 1);
```

Comment:
Previously the `Server` would keep track in `Server::active` whether any connection has events to process. On a call to `Server::active_connections` it would remove this state, whether the caller of `active_connections` actually consumed the events or not.

With this pull request, `Server` no longer keeps this additional state, i.e. `Sever::active` is removed. Instead of returning the cached information on `Server::active_connections`, the server simply asks the `Connection`s itself, whether they have events to process.

Given that this test consumes no events, the expected number of active connections is now `1` and not, as previously wrongly returned,  `0`.